torch/nn/modules/module.py

```diff
@@ -412,14 +417,16 @@ def to(self, *args, **kwargs):
 
         """
 
-        device, dtype, non_blocking = torch._C._nn._parse_to(*args, **kwargs)
+        device, dtype, non_blocking, convert_to_format = torch._C._nn._parse_to(*args, **kwargs)
 
         if dtype is not None:
             if not dtype.is_floating_point:
                 raise TypeError('nn.Module.to only accepts floating point '
                                 'dtypes, but got desired dtype={}'.format(dtype))
 
         def convert(t):
+            if convert_to_format is not None and t.dim() == 4:
+                return t.to(device, dtype if t.is_floating_point() else None, non_blocking, memory_format=convert_to_format)
```

Comment:
this doesn't match the documentation (which says the only case for `to` with memory-format is the 1-arg case).

Reply:
Pardon, but I read 'This can be called as' as: here examples of calls, but they are not limited to this options.

Reply:
that's not my reading of it, although I can see why you read it that way.

In particular, before memory_format was introduced, it corresponded exactly to the function signatures:
https://github.com/pytorch/pytorch/blob/66f2bba8527d941c7d73d3cae9e9576c601587a6/torch/csrc/autograd/utils/python_arg_parsing.h#L15-L17

(remove copy because it's not supported and remove memory_format because we are considering the case before memory_format was introduced).

So, the introduction of memory_format changed this from "these are the supported signatures" to "these are some examples of supported signatures".  I think the former is more useful and we should change it back.

Reply:
And actually, the example you added (`.. function:: to(memory_format=torch.channels_last)`) doesn't work because the parsing code hasn't been updated, right?

Reply:
IMO we should do the following:
1) Add a memory_format only-overload to the python parsing.
2) List the valid calls, i.e.:
```
        .. function:: to(device=None, dtype=None, non_blocking=False, memory_format=None)
        .. function:: to(dtype, non_blocking=False, memory_format=None)
        .. function:: to(tensor, non_blocking=False, memory_format=None)
        .. function:: to(memory_format)
```
(or similar).

Reply:
I had parsing updated https://github.com/pytorch/pytorch/pull/28991/files#diff-
7a05cfd8eb442889dffd6c3d2e4d0ddcR24 

Will update inline and html docs as follow-up PR